frontend/src/metabase/query_builder/actions/core/core.js

```diff
@@ -249,13 +251,15 @@ export const apiUpdateQuestion = (question, { rerunQuery } = {}) => {
       ? getQuestionWithDefaultVisualizationSettings(question, series)
       : question;
 
+    // Before we clean the query, we make sure question is not treated as a dataset
+    // as calling table() method down the line would bring unwanted consequences
```

Comment:
does this comment still make sense? I don't see `table()` method to be used

Reply:
Removed this